### ISSUE TITLE
Update to version 57

### ISSUE THREAD

**Issue by Eloston** [Eloston/ungoogled-chromium#176]
Here's what needs to be done (in approximate order):
* [x] ~Finish #125 (@Eloston is working on this)~
* [x] ~Include new patches from Debian (@Eloston is working on this)~
* [x] ~Include new patches from Inox (@Eloston is working on this)~
* [x] ~Include (or upgrade) patches from Iridium (@Eloston is working on this)~
* [x] ~Update ungoogled-chromium's own patches (@Eloston is working on this)~
* [x] ~Get it to build (on Debian Stretch first) (@Eloston is working on this)~
* [x] ~Refactor the build system again. Implement domain substitution in Debian build files generator (@Eloston is working on this)~
* [ ] Get builds on other platforms to work
  * [x] ~macOS support (Binaries will be published soon)~
  * [ ] Windows support (See #215)
* [ ] Work on other issues in milestone (Pending)
* [x] ~~Merge from `develop` to `master` (Pending)~~
* [x] ~~Finish upgrading downloads system (Pending)~~
* [ ] Publish binaries (Pending)

**NOTE**: From now on, development will occur on `develop` and then be merged into `master` once it is ready. `master` should be considered "somewhat stable", i.e. features aren't completely broken.

The other issues associated to the 56.x.x.x milestone that are not related to any of the above will be resolved afterwards. EDIT: New issues can be created for specific parts of the tasks as necessary (i.e. don't create an issue unless it ends up involving quite a bit of work or requires discussion)

**Comment by lenormf** [comment]
Will you experiment with community builds for this release? I have a `docker` container dedicated to building the project, so I can help on that front if needs be.

**Comment by Eloston** [comment]
@lenormf Thanks for offering. I'm not sure how the new system we're designing will work out in practice, so keep that on standby in case we need it.

**Comment by gcarq** [comment]
I'm currently compiling version 56.0.2924.76, some patches needed an update. The progress is being tracked here: https://github.com/gcarq/inox-patchset/issues/54

**Comment by Eloston** [comment]
@gcarq Alright, thanks for the heads up.

**Comment by triceratops1** [comment]
@eloston 
could you create "regional" issues
1 issue per issue

**Comment by Eloston** [comment]
@triceratops1 No, I will not create "regional issues" unless they are necessary. Please see the edit in the original post.

I appreciate your status updates related to the tasks. As a result, I have included additional status information for all of the tasks that I will try to keep up-to-date. This should remove the need for unnecessary notifications.

**Comment by triceratops1** [comment]
ok  http://mirrors.dcarsat.com.ar/debian/pool/main/c/chromium-browser/chromium-browser_56.0.2924.76-1.debian.tar.xz

**Comment by triceratops1** [comment]
iridium doesnt give updates

**Comment by triceratops1** [comment]
Include (or upgrade) patches from Iridium (Will wait for new Iridium patches unless the previous tasks are done), iridium doesnt update from 54 :-1: 


Update ungoogled-chromium's own patches (Pending) (you now've ready at linux=

**Comment by Eloston** [comment]
@triceratops1 I don't know what you're talking about. If you look at their git repo, they're currently in the process of updating to Chromium 57 (which is currently in beta).

So I will be updating them by hand to version 56 and include changes from the git repo as necessary.

**Comment by Eloston** [comment]
Status update: I have successfuly built version 56 for Debian stretch with the current changes in `develop`.

**Comment by nhantrn** [comment]
I don't want to make a new issue, so I'll report it here. For current release v 55.x

There seem to be some kind weird bug when loading www.ign.com that only happen only in ungoogled-chromium but not in other chrome variants. If you go to IGN.com, it'll connect to `www.example.com` and got redirected to `http://www.ign.com/g00/?i10c.referrer=` according to the network log. This does not happen in Chrome official. Nor in Firefox, IE, etc.

Also ublock does not block correctly and letting some ads slip through on IGN. However, if you open up the devtool, then the page will automatically reload and ublock will block all ads correctly. This only happens when you launch ungoogled chromium. After opening the devtool, ublock will work fine for the rest of the session.

**Comment by Eloston** [comment]
@nhantrn 

> I don't want to make a new issue, so I'll report it here. For current release v 55.x

Please do create a separate issue. Your bug is not related at all to this one. Why don't you want to create a separate issue? It's just a click of a button.

**Comment by Eloston** [comment]
Status update: I've updated the patches to version 57 and finished refactoring the new build system into `utilikit`. Adding support for other configurations and documenting the new build systems is up next.

**Comment by perfect7gentleman** [comment]
where is patch order for 57 ?

**Comment by Eloston** [comment]
@perfect7gentleman You can get the full patch order for a given configuration using `utilikit/export_resources.py` as it now says in BUILDING.md in `develop`. If you want patch orders for a specific configuration, you need to go into `resources/configs` and open up `patch_order` as documented in DESIGN.md

**Comment by perfect7gentleman** [comment]
@Eloston, okay.

**Comment by leeoniya** [comment]
will there be windows binaries for v57? i'd like to use it in combination with https://github.com/henrypp/chrlauncher and https://chromium.woolyss.com/

**Comment by Eloston** [comment]
@leeoniya Yes, there will be Windows binaries. I don't have an ETA because no one's working on updating the support right now.

**Comment by andrewoesten** [comment]
You need someone to make a Windows build and/or are there code changes needed to get it running? I might help out here.

**Comment by Eloston** [comment]
@andrewoesten No one's taken up the task to port to Windows yet. There may be some patch changes involved. There also should be some code for Windows written in `utilikit`.

**Comment by triceratops1** [comment]
@eloston  can you talk me about the flags

ad patch process?

for make a "human-readable" fork of ungoogled?

ideal for windows/mac/other distros (export to ebuild/pkgbuild/rpm-spec)

**Comment by Eloston** [comment]
@triceratops1 You should export the resources if you want to use them outside of utilikit. Read the documentation for more details.

**Comment by darkenvy** [comment]
Looking forward to being able to build for Mac OSX (v57 ofc). Is there a way to export on v57? The notes say only for linux and the build process doesn't work like it did on v55.

**Comment by Eloston** [comment]
@darkenvy If you mean export resources, then yes; that is all documented. There will be a pull request made soon with the changes to add macOS support.

**Comment by grandfroid** [comment]
Do you plan to realese the update to 57.xxx in .exe form ?

**Comment by Eloston** [comment]
@grandfroid No one is working on it right now, so I don't know when, if ever, it will be done.

**Comment by Eloston** [comment]
Also an update about macOS: Support has been added back in the `develop` branch right now, but I'll like to have at least two people try it out to see how well it works first (for now, @tectiv3 and @9Morello). If you'd like to build it and report back here, you're more than welcome to.

**Comment by tectiv3** [comment]
I'll give it a try later today.

**Comment by tectiv3** [comment]
It works.

**Comment by Driesje32** [comment]
Seems like as we're very close to release the project has come to hold a little bit... Any idea when the pre-built binaries will be released?

**Comment by 9Morello** [comment]
@Driesje32 I plan to release OS X pre-built binaries in the next days. Sorry for the delay.

**Comment by Eloston** [comment]
macOS binaries are up now.

**Comment by Eloston** [comment]
Development on 57 has ceased in favor of 58. See #228
